app/services/container_dashboard_service.rb

```diff
@@ -175,8 +173,54 @@ def ems_utilization
           :yData => ["used"] + used_mem.values.map { |m| (m / 1024.0).round }
         }
       }
-    else
-      {}
     end
   end
+
+  def hourly_network_trends
+    resource_ids = @ems.present? ? [@ems.id] : ManageIQ::Providers::ContainerManager.all.pluck(:id)
+    hourly_network_trend = Hash.new(0)
+    hourly_metrics =
+      MetricRollup.find_all_by_interval_and_time_range("hourly", 24.hours.ago.beginning_of_hour.utc, Time.now.utc)
+    hourly_metrics =
+      hourly_metrics.where('resource_type = ? AND resource_id in (?)', 'ExtManagementSystem', resource_ids)
+
+    hourly_metrics.each do |m|
+      hour = m.timestamp.beginning_of_hour.utc
+      hourly_network_trend[hour] += m.net_usage_rate_average
+    end
+
+    if hourly_metrics.any?
+      {
+        :xData => ["date"] + hourly_network_trend.keys,
+        :yData => ["used"] + hourly_network_trend.values.map(&:round)
+      }
+    end
+  end
+
+  def daily_network_trends
+    daily_network_trend = Hash.new(0)
+    daily_provider_metrics.each do |m|
+      day = m.timestamp.strftime("%Y-%m-%d")
+      daily_network_trend[day] += m.net_usage_rate_average
+    end
+
+    if daily_provider_metrics.any?
+      {
+        :xData => ["date"] + daily_network_trend.keys,
+        :yData => ["used"] + daily_network_trend.values.map(&:round)
+      }
+    end
+  end
+
+  def daily_provider_metrics
+    if @daily_metrics.blank?
+      resource_ids = @ems.present? ? [@ems.id] : ManageIQ::Providers::ContainerManager.all.pluck(:id)
+      @daily_metrics = VimPerformanceDaily.find_entries(:tz => @controller.current_user.get_timezone)
+      @daily_metrics =
+        @daily_metrics.where('resource_type = :type and resource_id in (:resource_ids) and timestamp > :min_time',
+                             :type => 'ExtManagementSystem', :resource_ids => resource_ids, :min_time => 30.days.ago)
+      @daily_metrics = @daily_metrics.order("timestamp")
+    end
+    @daily_metrics
+  end
```

Comment:
This is a singletonish sort of way to prevent this query from running twice. 
Feels un-rubylike. What do you think @moolitayer @simon3z ?